classes/output/index_page.php

```diff
@@ -37,47 +35,55 @@ class index_page implements \renderable, \templatable {
      *
      * @var array
      */
-    protected $selectedwebserviceindices = array();
+    protected $selectedwebserviceindices = [];
+
 
     /**
      * Constructor function - assign instance variable.
+     *
      * @param array $indicies
      */
     public function __construct(array $indicies) {
         $this->selectedwebserviceindices = $indicies;
+
     }
 
+
     /**
      * Exports the data for the index_page.mustache template
      *
-     * @param \renderer_base $output
+     * @param  \renderer_base $output
      * @return \stdClass
      */
-    public function export_for_template(\renderer_base $output): stdClass {
-        global $DB;
-
+    public function export_for_template(\renderer_base $output): object {
         // Return empty object if no selected webservices.
         if (empty($this->selectedwebserviceindices)) {
-            return new stdClass();
+            return (object) [];
         }
 
-        // Get_records returns an object array where key for each object is the name of the webservice.
-        // Use array_values to change key to the index of each object so that we can filter based on $selectedWebserviceIndices.
-        $webservicesrecords = array_values($DB->get_records('external_functions', array(), '', 'name'));
+        $exportwebservices = new \tool_wsformat\export_webservices('http', $this->selectedwebserviceindices);
```

Comment:
This constructor arg needs to be set to the real sitename. This is available in $CFG->wwwroot